lfsapi/client.go

```diff
@@ -275,7 +284,7 @@ func newRequestForRetry(req *http.Request, location string) (*http.Request, erro
 
 	for key := range req.Header {
 		if key == "Authorization" {
-			if req.URL.Scheme != newReq.URL.Scheme || req.URL.Host != newReq.URL.Host {
```

Comment:
I added some code in 36944dc in order to prevent insecure redirects. If a redirect is sent from HTTPS down to HTTP, the following error will be given:

```
lfsapi/client: refusing insecure redirect, https->http
```